task/test-failure-policy

```diff
@@ -0,0 +1,89 @@
+import subprocess
+import sys
+import unittest
+
+BASE = os.path.dirname(__file__)
+sys.path.insert(1, os.path.join(BASE, ".."))
+
+from lib.constants import BOTS_DIR
+
+
+PCP_CRASH = """
+# ----------------------------------------------------------------------
+# testFrameNavigation (check_multi_machine.TestMultiMachine)
+#
+Unexpected journal message '/usr/libexec/cockpit-pcp: bridge was killed: 11'
+not ok 110 testFrameNavigation (check_multi_machine.TestMultiMachine)
+Traceback (most recent call last):
+  File "test/verify/check-multi-machine", line 202, in tearDown
+    MachineCase.tearDown(self)
+  File "/home/martin/upstream/cockpit/test/common/testlib.py", line 533, in tearDown
+    self.check_journal_messages()
+  File "/home/martin/upstream/cockpit/test/common/testlib.py", line 689, in check_journal_messages
+    raise Error(first)
+Error: /usr/libexec/cockpit-pcp: bridge was killed: 11
+Wrote TestMultiMachine-testFrameNavigation-fedora-i386-127.0.0.2-2501-FAIL.png
+Wrote TestMultiMachine-testFrameNavigation-fedora-i386-127.0.0.2-2501-FAIL.html
+Wrote TestMultiMachine-testFrameNavigation-fedora-i386-127.0.0.2-2501-FAIL.js.log
+Journal extracted to TestMultiMachine-testFrameNavigation-fedora-i386-127.0.0.2-2501-FAIL.log
+Journal extracted to TestMultiMachine-testFrameNavigation-fedora-i386-127.0.0.2-2503-FAIL.log
+Journal extracted to TestMultiMachine-testFrameNavigation-fedora-i386-127.0.0.2-2502-FAIL.log
+"""
+
+
+class TestPolicy(unittest.TestCase):
+    def testKnownIssue(self):
+        script = os.path.join(BOTS_DIR, "test-failure-policy")
+        cmd = [script, "--offline", "example"]
+        proc = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, universal_newlines=True)
+        (output, unused) = proc.communicate(PCP_CRASH)
+        self.assertEqual(output, "Known issue #9876\n")
+
+    def testLineGlob(self):
+        script = os.path.join(BOTS_DIR, "test-failure-policy")
+        cmd = [script, "--offline", "example"]
+        proc = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, universal_newlines=True)
+        (output, unused) = proc.communicate("""
+> log: phase coils are misaligned by 0.34 micron
+> warning: this will explode in your face
+Some other mubo-jumbo
+Traceback (most recent call last):
+  File "test/verify/check-warp-drive", line 34, in testCoils
+     self.assertTrue(all_in_order)
+not ok 1 test/verify/check-warp-drive TestDrive.testCoils
+""")
+        self.assertEqual(output, "Known issue #123\n")
+
+        proc = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, universal_newlines=True)
+        (output, unused) = proc.communicate("""
+> log: phase coils are misaligned by 0.34 micron
+Traceback (most recent call last):
+  File "test/verify/check-warp-drive", line 34, in testCoils
+     self.assertTrue(all_in_order)
+not ok 1 test/verify/check-warp-drive TestDrive.testCoils
+""")
+        self.assertEqual(output, "Known issue #123\n")
```

Comment:
Please also add test cases for "do nothing" (one for a passing, one for a failing test), with empty output and exit code 0; and a "retry" case with code 1.

Reply:
done